streamparse/dsl/component.py

```diff
@@ -1,30 +1,177 @@
+                                        .format(input_spec))
+                    input_dict[stream_id] = grouping
+            elif inputs is None:
+                input_dict = {}
+            else:
+                raise TypeError('Inputs must either be a list, dict, or None.  '
+                                'Given: {!r}'.format(inputs))
+        if isinstance(config, dict):
+            config = json.dumps(config)
+        elif config is None:
+            config = '{}'
+        else:
+            raise TypeError('Config must either be a dict or None.  Given: {!r}'
+                            .format(config))
+        if outputs is None:
+            outputs = []
+        if isinstance(outputs, (list, tuple)):
+            streams = {}
+            for output in outputs:
+                if isinstance(output, Stream):
+                    streams[output.name] = StreamInfo(output_fields=output.fields,
+                                                      direct=output.direct)
+                # Strings are output fields for default stream
+                elif isinstance(output, string_types):
+                    default = streams.setdefault('default',
+                                                 StreamInfo(output_fields=[],
+                                                            direct=False))
+                    default.output_fields.append(output)
+                else:
+                    raise TypeError('Outputs must either be a list of strings '
+                                    'or a list of Streams.  Invalid entry: {!r}'
+                                    .format(output))
 
+        # Set validated/normalized arguments
         self.component_cls = component_cls
         self.name = name
         self.parallelism = parallelism
+        self.config = config
+        self.outputs = outputs
+        self.inputs = input_dict
+        self.common = ComponentCommon(inputs=input_dict, streams=streams,
+                                      parallelism_hint=parallelism,
+                                      json_conf=config)
+
+    def __getitem__(self, stream):
+        if stream not in self.common.streams:
+            raise KeyError('Invalid stream for {}: {!r}. Valid streams are: '
+                           '{!r}'.format(self.name, stream,
+                                         list(self.common.streams.keys())))
+        return GlobalStreamId(componentId=self.name, streamId=stream)
+
+    def __repr__(self):
+        """:returns: A string representation of the Specification. """
+        attr_dict = deepcopy(self.__dict__)
+        component_cls = attr_dict.pop('component_cls')
+        repr_str = '{}({cls}'.format(self.__class__.__name__,
+                                     cls=component_cls.__name__)
+        for key, val in attr_dict.items():
+            repr_str += ', {}={!r}'.format(key, val)
+        repr_str += ')'
+        return repr_str
+
+
+class JavaComponentSpec(ComponentSpec):
```

Comment:
is the idea going to be that if a `JavaComponentSpec` is in the topology, we build uberjars, but if not, we build simple jars?